Description: Add get_link_layer support to rdmavt
 Assign function pointer for ib_device get_link_layer.
Author: Brian T. Smith <bsmith@systemfabricworks.com>
Forwarded: not-needed
Last-Update: <2018-10-27>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/rdmavt/vt.c
+++ b/rdmavt/vt.c
@@ -343,6 +343,12 @@
 	return 0;
 }
 
+static enum rdma_link_layer rvt_get_link_layer(struct ib_device *device,
+											   u8 port_num)
+{
+	return IB_LINK_LAYER_INFINIBAND;
+}
+
 enum {
 	MISC,
 	QUERY_DEVICE,
@@ -388,6 +394,7 @@
 	RESIZE_CQ,
 	ALLOC_PD,
 	DEALLOC_PD,
+	GET_LINK_LAYER,
 	_VERB_IDX_MAX /* Must always be last! */
 };
 
@@ -711,6 +718,12 @@
 				      rvt_dealloc_pd);
 		break;
 
+	case GET_LINK_LAYER:
+		check_driver_override(rdi, offsetof(struct ib_device,
+											get_link_layer),
+							  rvt_get_link_layer);
+		break;
+
 	default:
 		return -EINVAL;
 	}
